### DISCUSSION TITLE
How should we handle boost metadata for bolt-11 (lnurl) payments

### DISCUSSION THREAD

**Discussion by Kolomona** [Podcastindex-org/podcast-namespace#674] [Ideas]
Payments sent using keysend have accompanying metadata sent via TLVs

This data is semi private as only a node receiving the payment can view the metadata.

I believe this is a good thing. When I send a boostagram I think of it as a message between myself and the recipients of the splits and not the entire world.

I propose that we put the metadata (boostagram) into a field of the bolt-11 invoice.
Ericpp has successfully tested this with this [pull request](https://github.com/ericpp/curiohoster/commit/c5080862601ce3644ae7685fd86873571361d5d5) to @thebells1111 curiohoster

![TLV in bolt-11](https://cdn.kolomona.com/media/images/v4v/tlv-in-bolt11.jpg)

If a person wishes to make their boostagrams public then adding a it to a 3rd party public protocol or platform should be trivial for the sending app.

There are many ways that this can be done. Here are a few that I have heard talked about

- Nostr
- ActivityPub
- Hive
- IRC 🙂

I believe it is **imperative** that participation with these public networks are on the part of the sender, **voluntary and opt-in**. 

As well as transparent. The booster should understand that they are making their boostagram public for all the world to see.

These are just my thoughts. I look forward to having this discussion with all you wonderful podcasting 2.0 nerd wizards. 🙂

## References

- [The 'lnaddress' Recipient Type](https://github.com/Podcastindex-org/podcast-namespace/blob/main/value/lnaddress.md)
- [Value Tag](https://github.com/Podcastindex-org/podcast-namespace/blob/main/value/value.md)
-  [TLV Registry](https://github.com/satoshisstream/satoshis.stream/blob/main/TLV_registry.md)
- [Add a new address type for lightning address - Discussion](https://github.com/Podcastindex-org/podcast-namespace/discussions/576)
- [BOLT 11 - Why is the description field limited to 639 bytes?](https://bitcoin.stackexchange.com/questions/90429/bolt-11-why-is-the-description-field-limited-to-639-bytes)

**Comment by ericpp** [comment]
It looks like Strike limits the comment length to 200 characters: https://strike.me/.well-known/lnurlp/adam_txfzikfutr . There's not much I can do with that particular field other than include a basic comment.

LNURL is very limited on the amount of data it accepts and wallets differ in the amount of spec they implement. Alby supports comments (LUD-12) https://github.com/lnurl/luds/blob/luds/12.md and payerData (LUD-18) https://github.com/lnurl/luds/blob/luds/18.md where as Strike only supports comments. Some other wallets may not even support comments.

TLVs could be sent like they were before if the wallet APIs supported including TLV records with invoice payments. Unfortunately, most of them seem to ignore that as an option even though LND supports it:

https://guides.getalby.com/developer-guide/alby-wallet-api/reference/api-reference/payments#bolt11-payment
https://www.webln.guide/building-lightning-apps/webln-reference/webln.sendpayment
https://github.com/nostr-protocol/nips/blob/master/47.md#pay_invoice
https://docs.strike.me/api/create-lightning-payment-quote


An intern at Alby even recognized a need for metadata with invoices in this blog post from 2 years ago: https://blog.getalby.com/making-bitcoin-transactions-smart-with-structured-metadata/

**Reply by ericpp** [reply]
With those restrictions, I think the most we could offer LNURL Bolt 11 users is a summary of their boost in the comment field with a way to obtain the full data from another service. Keysend users would be unaffected since we can continue sending TLVs to them as normal.
